torch/_inductor/codegen/cuda/gemm_template.py

```diff
@@ -660,7 +707,10 @@ def render(  # type: ignore[override]
         X, W = self.input_nodes[0], self.input_nodes[1]
         Y = self.output_node
         Bias = None if len(self.input_nodes) == 2 else self.input_nodes[2]
-
+        # The layouts might have changed between autotuning and this call if they were FlexibleLayout
+        # we need to adapt, which might lead to suboptimal performance.
+        # @TODO kadeng: Find a way to solve this better
```

Comment:
I believe, this `TODO` is sufficient :)

Reply:
Just FYI: This is addressed later in the diff stack..